File: bongf/make_prime_number.py
def solution(nums):
    answer = 0
    for x in range(len(nums) - 2):
        first = nums[x]
        for y in range(x+1, len(nums) - 1):
            second = nums[y]
            for z in range(y+1, len(nums)):
                third = nums[z]
                sum = first + second + third
                if is_prime(sum) : 
                    answer += 1
    return answer

def is_prime(number):
    half = int(number / 2)
    for i in range(2,half+1):
        if (number % i) == 0 :
            return False
    return True

def solution_2(number):
    from itertools import combinations as cb
    answer = 0
    for x in cb(number, 3) :
        sum_num = sum(x)
        if is_prime(sum_num):
            answer += 1
    return answer

def solution_3_is_prime(number):
    answer = 0
    for i in range(2, int(number**0.5)+1):
        if number % i == 0 :
            return 0
    return 1
    

def solution_3(number):
    from itertools import combinations as cb
    return sum( solution_3_is_prime(sum(x)) for x in cb(number, 3))

nums = [1,2,7,6,4]
print(solution_3(nums))

File: programmers/java/src/level1/MakePrimeNumber.java
package level1;

public class MakePrimeNumber {
    private int[] nums;
    private boolean[] selected;
    private int answer = 0;

    public int solution(int[] nums) {
        this.nums = nums;
        this.selected = new boolean[nums.length];
        select(0, 0);
        return answer;
    }

    private void select(int start, int cnt) {
        if (cnt == 3) {
            int sum = 0;
            for (int i = 0; i < selected.length; i++) {
                if (selected[i]) {
                    sum += nums[i];
                }
            }
            if (isPrime(sum)) {
                answer++;
            }
            return;
        }

        for (int i = start; i < selected.length; i++) {
            selected[i] = true;
            cnt += 1;
            select(i + 1, cnt);
            cnt -= 1;
            selected[i] = false;
        }

    }

    private boolean isPrime(int x) {
        if (x == 0) {
            return false;
        }
        if (x == 1) {
            return false;
        }
        for (int i = 2; i <= (int) Math.sqrt(x); i++) {
            if (x % i == 0) {
                return false;
            }
        }
        return true;
    }

    //I did with backtracking for combinations
    //But it's better to use tri-for? like bleow;
    public int solution2(int[] nums) {
        int ans = 0;

        for(int i = 0; i < nums.length - 2; i ++){
            for(int j = i + 1; j < nums.length - 1; j ++){
                for(int k = j + 1; k < nums.length; k ++ ){
                    if(isPrime(nums[i] + nums[j] + nums[k])){
                        ans += 1;
                    }
                }
            }
        }
        return ans;
    }
}
